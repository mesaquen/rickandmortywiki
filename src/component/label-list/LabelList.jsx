import React from 'react';
import PropTypes from 'prop-types';
import { Label } from '../../component';
/**
 * @author Mesaque Francisco <mesaquenf@gmail.com>
 * @since 2019-12-25
 */
const LabelList = props => {
  const { items } = props;

  const labelRenderer = label => (
    <Label
      key={label.title}
      title={label.title}
      data-testid={`label-${label.title}`}
    >
      {label.value}
    </Label>
  );

  return <>{items.map(labelRenderer)}</>;
};

LabelList.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default React.memo(LabelList);
