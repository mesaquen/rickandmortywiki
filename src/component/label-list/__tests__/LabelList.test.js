import React from 'react';
import { render, cleanup } from '@testing-library/react';
import LabelList from '../LabelList';
/**
 * @author Mesaque Francisco <mesaquenf@gmail.com>
 * @since 2019-12-25
 */
describe('LabelList', () => {
  const labelList = [
    { title: 'label01', value: 'value01' },
    { title: 'label02', value: 'value02' },
  ];

  afterEach(cleanup);

  it('should render without error', () => {
    let { getByText } = render(<LabelList items={labelList} />);
    const titleElement = getByText(labelList[0].title);
    const valueElement = getByText(labelList[0].value);
    expect(titleElement).toBeInTheDocument();
    expect(valueElement).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<LabelList items={labelList} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
