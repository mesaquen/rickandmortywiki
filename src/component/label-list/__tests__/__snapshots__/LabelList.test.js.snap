// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelList should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bxivhb kkzDWN"
  >
    <span
      class="sc-htpNat SwWzv"
      color="#999"
    >
      label01
    </span>
    <span
      class="sc-htpNat kNOgIk sc-ifAKCX gENqNG"
      color="teal"
    >
      value01
    </span>
  </div>
  <div
    class="sc-bxivhb kkzDWN"
  >
    <span
      class="sc-htpNat SwWzv"
      color="#999"
    >
      label02
    </span>
    <span
      class="sc-htpNat kNOgIk sc-ifAKCX gENqNG"
      color="teal"
    >
      value02
    </span>
  </div>
</DocumentFragment>
`;
